tests/testproject/TestProject.IntegrationServiceA/Program.cs

```diff
@@ -34,6 +34,10 @@
 {
     builder.AddRedisClient("redis");
 }
+if (!resourcesToSkip.HasFlag(TestResourceNames.garnet))
+{
+    builder.AddRedisClient("garnet");
```

Comment:
We might need to make these "keyed", since they both register IConnectionMultiplexer as singletons.

Reply:
ConnectionString are equal.
The values ​​from the first element are always taken.
![image](https://github.com/dotnet/aspire/assets/52016832/aa405a11-bb3b-4506-a8ae-547f4cbca947)
![image](https://github.com/dotnet/aspire/assets/52016832/59f197d1-9fc3-4d30-991e-ee5097ae5625)

Is this a bug or am I doing something wrong?

Reply:
This looks like something is going wrong. I would run the app with the dashboard and see what the environment variables are getting passed to the app. Look for ones that start with `ConnectionStrings__`

Reply:
Trying to debug it locally, it looks like a bug in the Redis component.

https://github.com/dotnet/aspire/blob/01c625a518ad2b3a438a3607e122873c6395a50e/src/Components/Aspire.StackExchange.Redis/AspireRedisExtensions.cs#L83-L89

This needs to respect the "serviceKey". It is adding an `OptionsFactory<ConfigurationOptions>` without the key, which means it will be global.

I've opened https://github.com/dotnet/aspire/issues/3873 for this.

Reply:
What do we end up doing with this? Are we waiting for a fix #3873 ?

Reply:
> What do we end up doing with this? Are we waiting for a fix #3873 ?

That issue is fixed in main.